compiler/rustc_interface/src/passes.rs

```diff
@@ -884,7 +884,11 @@ fn analysis(tcx: TyCtxt<'_>, cnum: CrateNum) -> Result<()> {
 
     sess.time("MIR_effect_checking", || {
         for def_id in tcx.body_owners() {
-            mir::transform::check_unsafety::check_unsafety(tcx, def_id);
+            if tcx.sess.opts.debugging_opts.thir_unsafeck {
+                tcx.ensure().thir_check_unsafety(def_id);
+            } else {
+                mir::transform::check_unsafety::check_unsafety(tcx, def_id);
```

Comment:
My one concern about landing this PR is that we are no longer testing the MIR unsafety check! Can we either (a) make this flag run *both* unsafety checkers or (b) run the tests in two modes using the [revisions](https://rustc-dev-guide.rust-lang.org/tests/adding.html?highlight=revisions#revisions) support.

Reply:
> (b) run the tests in two modes using the revisions support.

Isn't this what I'm already doing? Or am I doing it wrong?

Reply:
```rust
// revisions: mir thir
// [thir]compile-flags: -Z thir-unsafeck
```

Reply:
Oh, did I miss that?